vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataController.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.component.grid.dataview;
+
+import java.util.ArrayList;
+import java.util.stream.Stream;
+
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.data.provider.DataCommunicator;
+import com.vaadin.flow.data.provider.DataController;
+import com.vaadin.flow.data.provider.DataKeyMapper;
+import com.vaadin.flow.data.provider.DataProvider;
+import com.vaadin.flow.data.provider.SizeChangeEvent;
+import com.vaadin.flow.data.provider.SizeChangeListener;
+import com.vaadin.flow.shared.Registration;
+
+/**
+ * DataController for Grid data communication.
+ *
+ * @param <T>
+ *         item type
+ * @since
+ */
+public class GridDataController<T> implements DataController<T> {
```

Comment:
I'm not yet sure if this is the correct way to do it, but probably fine to leave for now as we still need to pull things like the size change listener to somewhere else.